src/kernels/composable_kernel/include/utility/in_memory_operation.hpp

```diff
@@ -83,6 +83,20 @@ struct SetData
 #endif
 };
 
+template <index_t DataPerAccess>
+struct SetData<int, DataPerAccess>
```

Comment:
This looks the same as the original template, why specializing it ?

Reply:
Without this patch,  the Reduction will report "device codes not found" failure.  It is because in the case CK_USE_AMD_BUFFER_ADDRESSING is defined,  amd_buffer_load & amd_buffer_store is used to transfer data, but they are not specialized for "int".   Reduction needs to transfer "indices"  using Blockwise/Threadwise_generic_copy.

Reply:
Then you would need to implement buffer_load/store for ```int``` type. They are much better than global_load/store in performance, and it reduce code divergence.

It should be easy
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/46aa3d8ac5b569dfd537be5ae3fc35b694224c95/src/kernels/composable_kernel/include/utility/amd_buffer_addressing.hpp#L172-L175

You can reuse that function for ```float``` to load ```int```
```
template <> 
 __device__ int amd_buffer_load<int, 1>(const int* p_src_block, 
                                            index_t src_thread_data_offset, 
                                            index_t src_const_data_offset) 
{
      const float src_float = amd_buffer_load<float, 1> (reinterpret_cast<const float*>(p_src_block, src_thread_data_offset, src_const_data_offset);
      return *(reinterpert_cast<const int*>(&src_float));
}
```

Same thing for buffer_store

Reply:
Check https://github.com/ROCmSoftwarePlatform/MIOpen/pull/322/commits/b14f6f05b0a360a317186d15f0c342e9ffe78bc9. I stay using the SetData<int, ...> specialization. Don't want the patch be too intrusive and don't have to define vector_type for integer type.

Reply:
Again, check https://github.com/ROCmSoftwarePlatform/MIOpen/pull/322/commits/5feabf6dcf47a51a6deb51cc5d3863390c69cf58

Reply:
Ok, that also works